p2p/src/sync/peer_common/mod.rs

```diff
@@ -124,11 +124,17 @@ pub fn choose_peers_best_block(
         (None, None) => Ok(None),
         (Some(id), None) | (None, Some(id)) => Ok(Some(id)),
         (Some(old_id), Some(new_id)) => {
-            let old_height =
-                chainstate.get_block_height_in_main_chain(&old_id)?.unwrap_or(0.into());
-            let new_height =
-                chainstate.get_block_height_in_main_chain(&new_id)?.unwrap_or(0.into());
-            if new_height >= old_height {
+            let old_block_chain_trust = chainstate
+                .get_gen_block_index(&old_id)?
+                .map_or(Uint256::ZERO, |idx: chainstate::GenBlockIndex| {
+                    idx.chain_trust()
+                });
+            let new_block_chain_trust = chainstate
+                .get_gen_block_index(&new_id)?
+                .map_or(Uint256::ZERO, |idx: chainstate::GenBlockIndex| {
+                    idx.chain_trust()
+                });
+            if new_block_chain_trust >= old_block_chain_trust {
```

Comment:
1. The explicit type on the closures' parameters is not needed.
2. The documentation for this function is outdated - it still mentions "bigger height" and the block being "on the main chain".

Reply:
Done